src/t_set.c

```diff
@@ -895,16 +895,16 @@ void sinterGenericCommand(client *c, robj **setkeys,
     if (dstkey) {
         /* Store the resulting set into the target, if the intersection
          * is not an empty set. */
-        int deleted = dbDelete(c->db,dstkey);
         if (setTypeSize(dstset) > 0) {
-            dbAdd(c->db,dstkey,dstset);
+            setKey(c,c->db,dstkey,dstset);
+            decrRefCount(dstset);
```

Comment:
these two `decrRefCount` can merge and move out of if ... else scope.

Reply:
Thanks, fixed.